src/Helper/GridHelperService.php

```diff
@@ -943,20 +944,25 @@ private function optimizedConcatLike(string $fullpath): string
      * A more performant alternative to "CONCAT(`path`,`key`) NOT LIKE $fullpath"
      * Set $onlyChildren to true when you want to exclude the folder/element itself
      */
-    private function optimizedConcatNotLike(string $fullpath, bool $onlyChildren = false): string
+    private function optimizedConcatNotLike(
+        string $fullpath,
+        bool $onlyChildren = false,
+        string $type = 'object'
+    ): string
     {
         $pathParts = explode('/', $fullpath);
         $leaf = array_pop($pathParts);
         $path = implode('/', $pathParts);
+        $queryColumn = $type === 'asset' ? '`filename`' : '`key`';
 
         if ($onlyChildren) {
             return '`path` NOT LIKE "' . $fullpath . '/%"';
         }
 
         return '(
-            (`path` != "' . $path . '/" AND `key` != "' . $leaf . '")
+            (`path` != "' . $path . '/" AND ' . $queryColumn .  ' != "' . $leaf . '")
             AND
-            `path` NOT LIKE "' . $fullpath . '/%"
+            `path` NOT LIKE "' . $fullpath . '%"
```

Comment:
I don't think we need that change, right? 
It should not make any difference, but I would prefer to remove it and keep `"' . $fullpath . '/%"` as this prevents including unwanted folders. 

Example constellation would be: 
`foo/bar`
`foo/barmore` 

With old query, `foo/barmore` is not affected, with new query it is. 

As said, currently it does not make any difference, but it could bite us in future cases. 

I will do another PR for `optimizedConcatLike` to optimize that as well.

Reply:
reverted it now, so we can proceed. 
thx very much for your support!!